core/watcher/eventsource/multiwatcher.go

```diff
@@ -0,0 +1,142 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package eventsource
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v4"
+	"github.com/juju/worker/v4/catacomb"
+)
+
+// Applier is a function that applies a change to a value.
+type Applier[T any] func(T, T) T
+
+// MultiWatcher implements Watcher, combining multiple Watchers.
+type MultiWatcher[T any] struct {
+	catacomb         catacomb.Catacomb
+	staging, changes chan T
+	applier          Applier[T]
+}
+
+// NewMultiNotifyWatcher creates a NotifyWatcher that combines
+// each of the NotifyWatchers passed in. Each watcher's initial
+// event is consumed, and a single initial event is sent.
+func NewMultiNotifyWatcher(ctx context.Context, watchers ...Watcher[struct{}]) (*MultiWatcher[struct{}], error) {
+	applier := func(_, _ struct{}) struct{} {
+		return struct{}{}
+	}
+	return NewMultiWatcher[struct{}](ctx, applier, watchers...)
+}
+
+// NewMultiStringsWatcher creates a StringsWatcher that combines
+// each of the StringsWatcher passed in. Each watcher's initial
+// event is consumed, and a single initial event is sent.
+func NewMultiStringsWatcher(ctx context.Context, watchers ...Watcher[[]string]) (*MultiWatcher[[]string], error) {
+	applier := func(current, additional []string) []string {
+		return append(current, additional...)
```

Comment:
This is not used yet, but I suspect it should probably have set-like behaviour.

Reply:
I've removed it, no point having deadcode. Also we don't know how it would be used. There are different variations for strings.